securitycenter/snippets_v2/snippets_findings_v2.py

```diff
@@ -0,0 +1,309 @@
+  # location_id= "global"
+  all_sources = f"{parent}/sources/{source_id}/locations/{location_id}"
+
+  group_result_iterator = client.group_findings(
+      request={
+          "parent": all_sources,
+          "group_by": "category",
+          "filter": 'state="ACTIVE"',
+      }
+  )
+  for count, group_result in enumerate(group_result_iterator):
+    print((count + 1), group_result)
+  # [END securitycenter_group_filtered_findings_v2]
+  return count
+
+
+def list_findings_with_security_marks(organization_id, source_id, location_id):
+  count = 0
+  # [START securitycenter_list_findings_with_security_marks_v2]
+  from google.cloud import securitycenter_v2
+
+  # Create a new client.
+  client = securitycenter_v2.SecurityCenterClient()
+
+  # 'parent' must be in one of the following formats:
+  #   "organizations/{organization_id}"
+  #   "projects/{project_id}"
+  #   "folders/{folder_id}"
+  parent = f"organizations/{organization_id}"
+
+  # 'source_id' to scope the findings.
+  # The "sources/-" suffix lists findings across all sources.  You
+  # also use a specific source_name instead.
+  # location_id= "global"
+  all_sources = f"{parent}/sources/{source_id}/locations/{location_id}"
+  finding_result_iterator = client.list_findings(
+      request={"parent": all_sources, "filter": 'NOT security_marks.marks.ACK="true" AND NOT mute="MUTED" AND state="ACTIVE"'}
+  )
+  # Iterate an print all finding names and the resource they are
+  # in reference to.
+  for count, finding_result in enumerate(finding_result_iterator):
+    print(
+        "{}: name: {} resource: {}".format(
+            count, finding_result.finding.name, finding_result.finding.resource_name
+        )
+    )
+  # [END securitycenter_list_findings_with_security_marks_v2]
+  return count
+
+
+def group_findings_by_state(organization_id, source_id, location_id):
+  """Demonstrates grouping all findings across an organization."""
+  count = 0
+  # [START securitycenter_group_findings_by_state_v2]
+  from google.cloud import securitycenter_v2
+
+  # Create a client.
+  client = securitycenter_v2.SecurityCenterClient()
+
+  # 'parent' must be in one of the following formats:
+  #   "organizations/{organization_id}"
+  #   "projects/{project_id}"
+  #   "folders/{folder_id}"
+  parent = f"organizations/{organization_id}"
+
+  # 'source_id' to scope the findings.
+  # The "sources/-" suffix lists findings across all sources.  You
+  # also use a specific source_name instead.
+  # location_id= "global"
+  all_sources = f"{parent}/sources/{source_id}/locations/{location_id}"
+
+  group_result_iterator = client.group_findings(
+      request={"parent": all_sources, "group_by": "state"}
+  )
+  for count, group_result in enumerate(group_result_iterator):
+    print((count + 1), group_result)
+  # [END securitycenter_group_findings_by_state_v2]
+  return count
+
+
+
+def create_source(organization_id):
+  """Create a new findings source."""
+  # [START securitycenter_create_source]
```

Comment:
All the source samples exists here: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/2ed1148d0afdae3a852282d71392bb511cbdb123/securitycenter/snippets/snippets_findings.py#L45

Kindly do not duplicate the samples. Reuse the existing ones and remove "Create source, list, update and get source" from this file.

Reply:
addressed